File: interview-coach-backend/schemas/analytics.py
from pydantic import BaseModel, Field


class ScoreTrendResponse(BaseModel):
    scores: list[float] = Field(default_factory=list)
    dates: list[str] = Field(default_factory=list)


class DashboardSummaryResponse(BaseModel):
    resume_improvement: float       
    interview_improvement: float
    job_match_improvement: float
    total_reports: int
    latest_resume_score: float | None = None
    latest_interview_score: float | None = None
    latest_job_match_score: float | None = None

File: interview-coach-frontend/src/types/analytics.ts
export interface ScoreTrendResponse {
  scores: number[];
  dates: string[];
}

export interface DashboardSummaryResponse {
  resume_improvement: number;
  interview_improvement: number;
  job_match_improvement: number;
  total_reports: number;
  latest_resume_score: number | null;
  latest_interview_score: number | null;
  latest_job_match_score: number | null;
}